server/src/main/java/io/druid/server/http/CoordinatorDynamicConfigsResource.java

```diff
@@ -75,20 +75,18 @@ public Response getDynamicConfigs()
   // default value is used for backwards compatibility
   @POST
   @Consumes(MediaType.APPLICATION_JSON)
+  @Produces(MediaType.APPLICATION_JSON)
   public Response setDynamicConfigs(final CoordinatorDynamicConfig dynamicConfig,
                                     @HeaderParam(AuditManager.X_DRUID_AUTHOR) @DefaultValue("") final String author,
                                     @HeaderParam(AuditManager.X_DRUID_COMMENT) @DefaultValue("") final String comment,
                                     @Context HttpServletRequest req
   )
   {
-    if (!manager.set(
+    return manager.set(
         CoordinatorDynamicConfig.CONFIG_KEY,
         dynamicConfig,
         new AuditInfo(author, comment, req.getRemoteAddr())
-    )) {
-      return Response.status(Response.Status.BAD_REQUEST).build();
-    }
-    return Response.ok().build();
+    ) ? Response.ok().build() : Response.status(Response.Status.BAD_REQUEST).build();
```

Comment:
actually, it looks like it doesn't catch serde error @ `io/druid/common/config/ConfigManager.java:170`